src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -236,6 +239,13 @@ internal static async Task HandleExceptionAsync(TimeoutAttribute timeout, Except
                 // Create a linked token source that will allow us to signal function cancellation
                 // (e.g. Based on TimeoutAttribute, etc.)                
                 CancellationTokenSource functionCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+                if (_drainModeManager.DrainModeEnabled)
+                {
+                    functionCancellationTokenSource.Cancel();
+                    logger?.LogInformation($"Requesting cancellation for function invocation:{instance.Id}");
+                }
```

Comment:
So this will cancel any new functions before they're invoked. But what about functions that are currently running? Do we need to cancel those as well?

Reply:
For currently running executions the data role / dwas would wait longer that the function timeout period for that SKU before scaling in. This would give currently running functions enough time to finish execution. So we shouldn't need to cancel currently running functions.

Reply:
We can't rely on that infrastructure being present in standalone WebJobs scenarios. We need to ensure those are also signaled.